File: rental-backend/src/main/java/com/realestate/rental/dto/AnalyticsDto.java
package com.realestate.rental.dto;

import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.Map;

@Data
@Builder
public class AnalyticsDto {
    private Long totalUsers;
    private Long totalTenants;
    private Long totalLandlords;
    private Long totalProperties;
    private Long pendingProperties;
    private Long approvedProperties;
    private Long totalBookings;
    private Long pendingBookings;
    private Long approvedBookings;
    private Long totalMessages;
    private Map<String, Long> propertiesByType;
    private Map<String, Long> propertiesByCity;
    private Map<String, Long> bookingsByStatus;
    private List<TopPropertyDto> topViewedProperties;
    private List<TopPropertyDto> topFavoritedProperties;
}

File: rental-app-flutter/lib/features/admin/models/analytics_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'analytics_dto.freezed.dart';
part 'analytics_dto.g.dart';

@freezed
abstract class AnalyticsDto with _$AnalyticsDto {
  const factory AnalyticsDto({
    @Default(0) int totalUsers,
    @Default(0) int totalTenants,
    @Default(0) int totalLandlords,
    @Default(0) int totalProperties,
    @Default(0) int pendingProperties,
    @Default(0) int approvedProperties,
    @Default(0) int totalBookings,
    @Default(0) int pendingBookings,
    @Default(0) int approvedBookings,
    @Default(0) int totalMessages,
    Map<String, int>? propertiesByType,
    Map<String, int>? propertiesByCity,
    Map<String, int>? bookingsByStatus,
  }) = _AnalyticsDto;

  factory AnalyticsDto.fromJson(Map<String, dynamic> json) => _$AnalyticsDtoFromJson(json);
}
